src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ICSharpBinder.cs

```diff
@@ -10,19 +10,19 @@ namespace Microsoft.CSharp.RuntimeBinder
     internal interface ICSharpBinder
     {
         CSharpArgumentInfo GetArgumentInfo(int index);
-        Type CallingContext { get; }
-        bool IsChecked { get; }
 
         // This is true for any binder that is eligible to take value type receiver
         // objects as a ref (for mutable operations). Such as calls ("v.M(d)"),
         // and indexers ("v[d] = v[d]"). Note that properties are not here because they
         // are only dispatched dynamically when the receiver is dynamic, and hence boxed.
         bool IsBinderThatCanHaveRefReceiver { get; }
 
-        void PopulateSymbolTableWithName(SymbolTable symbolTable, Type callingType, ArgumentObject[] arguments);
+        void PopulateSymbolTableWithName(Type callingType, ArgumentObject[] arguments);
 
         Expr DispatchPayload(RuntimeBinder runtimeBinder, ArgumentObject[] arguments, LocalVariableSymbol[] locals);
```

Comment:
Just wonder - how large is the `RuntimeBinder` now? Should it be passed as an `in` parameter?   (not in this change, just a thought..)

Reply:
It's a reference and a bool

Reply:
Yes. It contains other structs but eventually it is indeed just the BindingContext. 
Such small struct is unlikely to benefit from `in`. Actually with byval there are more chances that JIT can do something smart - promote into scalars, pass in registers, etc..

---
In reply to: [168586125](https://github.com/dotnet/corefx/pull/27044#discussion_r168586125) [](ancestors = 168586125)

Reply:
That was my reasoning, and it's not passed often enough per operation to make it worth profiling just to see if something unexpected makes `in` pay off, IMO.